### DISCUSSION TITLE
[code] config file and options

### DISCUSSION THREAD

**Discussion by artesim-zz** [Ortho4XP/Ortho4XP#23] [Ideas]
I have positive experience with ConfigArgParse module, which allow to define a list of options once (derives from the standard argparse => same syntax, same doc), and have the configuration resolved from, in order :
- command line options
- environment variables (not a fan, but possible)
- config file (several locations possible)
- hard-coded default values

**Comment by girotobial** [comment]
I was thinking along the lines of Pydantic, haven't tried ConfigArgParse before. 

I think it would be worth mapping out what the requirements are for config management are, before we pick a solution.

**Reply by artesim-zz** [reply]
Oh, this also looks very interesting indeed.
At a glance, I didn't see anything about config files as a source, does it support it ?

**Reply by girotobial** [reply]
So it's a bit buried but under the Settings management there is the ability to add sources. [https://pydantic-docs.helpmanual.io/usage/settings/#adding-sources](https://pydantic-docs.helpmanual.io/usage/settings/#adding-sources)
